src/Microsoft.PowerShell.Commands.Utility/commands/utility/Csv.cs

```diff
@@ -20,92 +22,69 @@ internal CSVHelper(char delimiter)
         /// </summary>
         internal char Delimiter { get; } = ',';
 
+        internal char Quote { get; } = '"';
+
         /// <summary>
         /// Parse a CSV string.
         /// </summary>
         /// <param name="csv">
         /// String to be parsed.
         /// </param>
         internal Collection<string> ParseCsv(string csv)
-        {
+        {        
+            csv = csv.Trim(); 
             Collection<string> result = new Collection<string>();
-            string tempString = string.Empty;
-            csv = csv.Trim();
             if (csv.Length == 0 || csv[0] == '#')
             {
                 return result;
             }
 
-            bool inQuote = false;
-            for (int i = 0; i < csv.Length; i++)
+            var reader = new StringReader(csv);
+            StringBuilder stringBuilder = new StringBuilder();
+
+            while (reader.Peek() != -1) 
             {
-                char c = csv[i];
-                if (c == Delimiter)
+                char nextChar = (char)reader.Read();
+
+                // if next character was delimiter, add string to builder and clear builder
```

Comment:
This comment does not seem to reflect what the code does, I assume you meant `add string to result and clear builder`.